### ISSUE TITLE
Updated syntax for multinode usage

### ISSUE THREAD

**Issue by matt-gss** [webdataset/webdataset#233]
Hi, I'm working on Google Colab and trying to setup minimal example of multi-core pytorch training with webdataset using data on GCP Buckets. Specifically I've got a bucket with my training shards and another bucket with test shards.

For colab setup, I'm using the suggested wheels from the pytorch XLA notebook for multi-core and adding webdataset to the pip installs. I see I've got webdataset 0.2.31 from pip show webdataset in the console.

I tried to start from (https://webdataset.github.io/webdataset/multinode/) with this code in the map function:
``` 
urls = list(braceexpand.braceexpand(FLAGS['train_urls']))
dataset = (wds.WebDataset(urls)
             .to_tuple('x_data.npy', 'y_data.npy')
             .map(preprocess)
             .batched(50))
```
but I'm seeing some strange behavior that I don't understand. Specifically, the docs indicates that a default nodesplitter and split_by_worker should be in effect:

```
urls = list(braceexpand.braceexpand("dataset-{000000..000999}.tar"))
dataset = wds.ShardList(urls, splitter=wds.split_by_worker, nodesplitter=wds.split_by_node, shuffle=False)
dataset = wds.Processor(dataset, wds.url_opener)
dataset = wds.Processor(dataset, wds.tar_file_expander)
dataset = wds.Processor(dataset, wds.group_by_keys)
```

What I see from the following however is that every core processes every file with the following in my map function. (Next step would be to investigate worker splitting once we've got shards distributed across cores)
```
loader = DataLoader(dataset, num_workers=FLAGS['num_workers']) 
for batch, (x,y) in enumerate(loader):
  print(f'core:{rank} batch:{batch}: x_min: {torch.min(x)} x_max: {torch.max(x)} len:{x.size()}')
```
I saw a related post with a suggestion to use v2 branch (#https://github.com/webdataset/webdataset/issues/138) but (if possible) I'd like to have my dependency be the main release that's going to come in under webdataset with pip.

I also attempted to expand on the wds.Processor approach but that doesn't seem to be available in this version maybe? High level just looking for best advice on implementing a pretty vanilla version of this with shard shuffling and sample shuffling. I'm not sure how shard shuffling across cores is coordinated (do you use a parallel loader?) so thoughts on that would be great too. If a minimum working example notebook would improve/clarify this question that's also something I'm happy to provide.

**Comment by jrcavani** [comment]
Just in case you are still trying to fix this, I was able to get the right split behavior by using what was suggested. A simple visual test:

```python
import os
import webdataset as wds
import torch
import torch.distributed as dist

if __name__ == "__main__":
    try:
        rank = int(os.environ["RANK"])
        world_size = int(os.environ["WORLD_SIZE"])
        local_rank = int(os.environ["LOCAL_RANK"])
        dist.init_process_group("nccl")
    except KeyError:
        rank = 0
        world_size = 1
        local_rank = 0
        dist.init_process_group(
            backend="nccl",
            init_method="tcp://127.0.0.1:12584",
            rank=rank,
            world_size=world_size,
        )

    dataset = wds.DataPipeline(
        wds.SimpleShardList("source-{000000..000999}.tar"),
        wds.split_by_node,
        wds.split_by_worker,
    )

    for idx, i in enumerate(iter(dataset)):
        if idx == 2:
            break

        print(f"rank: {rank}, world_size: {world_size}, {i}")
```

```
--> torchrun --nproc-per-node 8 test-webdataset-dist.py
WARNING:torch.distributed.run:
*****************************************
Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed.
*****************************************
rank: 3, world_size: 8, {'url': 'source-000003.tar'}
rank: 3, world_size: 8, {'url': 'source-000011.tar'}
rank: 1, world_size: 8, {'url': 'source-000001.tar'}
rank: 1, world_size: 8, {'url': 'source-000009.tar'}
rank: 6, world_size: 8, {'url': 'source-000006.tar'}
rank: 6, world_size: 8, {'url': 'source-000014.tar'}
rank: 5, world_size: 8, {'url': 'source-000005.tar'}
rank: 5, world_size: 8, {'url': 'source-000013.tar'}
rank: 0, world_size: 8, {'url': 'source-000000.tar'}
rank: 0, world_size: 8, {'url': 'source-000008.tar'}
rank: 4, world_size: 8, {'url': 'source-000004.tar'}
rank: 4, world_size: 8, {'url': 'source-000012.tar'}
rank: 7, world_size: 8, {'url': 'source-000007.tar'}
rank: 7, world_size: 8, {'url': 'source-000015.tar'}
rank: 2, world_size: 8, {'url': 'source-000002.tar'}
rank: 2, world_size: 8, {'url': 'source-000010.tar'}
```

**Comment by jrcavani** [comment]
And if you do shuffle, use `detshuffle`,

```python
    dataset = wds.DataPipeline(
        wds.SimpleShardList("source-{000000..000016}.tar"),
        wds.shuffle(),
        wds.split_by_node,
        wds.split_by_worker,
    )
```

```
rank: 7, world_size: 8, {'url': 'source-000015.tar'}
rank: 7, world_size: 8, {'url': 'source-000014.tar'}
rank: 4, world_size: 8, {'url': 'source-000008.tar'}
rank: 4, world_size: 8, {'url': 'source-000010.tar'}
rank: 5, world_size: 8, {'url': 'source-000007.tar'}
rank: 2, world_size: 8, {'url': 'source-000000.tar'}
rank: 5, world_size: 8, {'url': 'source-000016.tar'}
rank: 2, world_size: 8, {'url': 'source-000009.tar'}
rank: 0, world_size: 8, {'url': 'source-000012.tar'}
rank: 0, world_size: 8, {'url': 'source-000011.tar'}
rank: 1, world_size: 8, {'url': 'source-000013.tar'}
rank: 1, world_size: 8, {'url': 'source-000005.tar'}
rank: 6, world_size: 8, {'url': 'source-000006.tar'}
rank: 6, world_size: 8, {'url': 'source-000003.tar'}
rank: 3, world_size: 8, {'url': 'source-000004.tar'}
rank: 3, world_size: 8, {'url': 'source-000001.tar'}
```

If you use `shuffle()` there, each rank/dataloader worker would get different random seed and shuffle non-deterministically.

**Comment by jrcavani** [comment]
With regard to the `WebDataset` API I was able to get the correct splitting,

```python
    dataset = (
        wds.WebDataset("source-{000000..00007}.tar", shardshuffle=True, detshuffle=True, nodesplitter=wds.split_by_node)
    )
    dataset.pipeline.pop()

    for idx, i in enumerate(iter(dataset)):
        print(f"rank: {rank}, world_size: {world_size}, {i}")
```

```
rank: 0, world_size: 8, {'url': 'source-000000.tar'}
rank: 1, world_size: 8, {'url': 'source-000001.tar'}
rank: 2, world_size: 8, {'url': 'source-000002.tar'}
rank: 3, world_size: 8, {'url': 'source-000003.tar'}
rank: 4, world_size: 8, {'url': 'source-000004.tar'}
rank: 5, world_size: 8, {'url': 'source-000005.tar'}
rank: 6, world_size: 8, {'url': 'source-000006.tar'}
rank: 7, world_size: 8, {'url': 'source-000007.tar'}
```

I'm testing using version 0.2.57.

**Comment by tmbdev** [comment]
WebDataset does not perform node splitting by default, just splitting by workers. It is supposed to output an error when it discovers that it is running multinode, but doesn't seem to be doing it here.

I'll see whether I can improve the error detection and docs.
